### PR TITLE
Build: Adds an editorconfig

### PR DESCRIPTION
**Pull request by ebenoist**
Adds an editorconfig file to establish minimal standards for edits to any project file

### CODE REVIEW

**Comment by leobalter** [comment]
This should follow the discussion on https://github.com/jquery/api.qunitjs.com/pull/114#issuecomment-150332207

**Comment by leobalter** [comment]
Also: jquery/contribute.jquery.org#129

**Comment by jzaefferer** [comment]
Though this should be fine to land as-is now, we can improve it later.

**Comment by arthurvr** [comment]
The package.json file uses 2 space indentation.

**Comment by leobalter** [comment]
The package.json is mostly updated through npm, which always rewrites it after a `npm install --save` or `npm version`.

We may follow the suggestion on the [editorconfig website](http://editorconfig.org/):

```yaml
# Matches the exact files either package.json or .travis.yml
[{package.json,.travis.yml}]
indent_style = space
indent_size = 2
```

**Comment by ebenoist** [comment]
I went ahead and added rules for *.json and *.yml files. Let me know if I should add anything else.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+root = true
+
+[*]
+indent_style = tab
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{json,yml}]
```
Comment (jzaefferer) [inline .editorconfig]: This should be specific to the affected files, `[{package.json,.travis.yml}]`.

Comment (ebenoist) [inline .editorconfig]: I can possibly see that for *.json, but shouldn't yml _always_ be two space indented?

Comment (jzaefferer) [inline .editorconfig]: You're right, that's in the spec: http://yaml.org/spec/current.html#id2519916 - apparently the amount doesn't matter, but sticking with `2` makes sense.

**Comment by scottgonzalez** [comment]
Let's not let the larger discussion hold this up. @leobalter is there anything stopping this from being merged?

**Comment by leobalter** [comment]
It LGTM.

**Pull request closed without merging**